### ISSUE TITLE
unexpected color palette behavior

### ISSUE THREAD

**Issue by allaway** [griffithlab/GenVisR#302]
Hi there,

Perhaps I am simply doing something wrong, but mainPalette doesn't behave as I would expect. For example, if I try to color a plot with 7 mutations types present using the following code:

`cols <- c("blue", "red", "pink", "green", "orange", "black", "brown")`

`waterfall(soms, fileType = "MAF", mainRecurCutoff = 0.3, mainDropMut = TRUE, mainPalette = cols)`

I end up with a plot that has variants colored with a blue-red-green gradient, rather than the 7 explicit colors that are in `cols`. Why is this?

Thanks for the great package though - very nice for visualizing mafs!

**Comment by zlskidmore** [comment]
Hi @allaway that is strange, I can't seem to reproduce it with the test data:
```
cols <- c("blue", "red", "pink", "green", "orange", "black", "brown")
waterfall(brcaMAF, fileType = "MAF", mainRecurCutoff = 0.1, mainDropMut = TRUE, mainPalette = cols)
```
our your mutations of type numeric by chance? you can check this by setting the out param.
```
output <- waterfall(brcaMAF, fileType = "MAF", mainRecurCutoff = 0.1, mainDropMut = TRUE, mainPalette = cols, out="data")
class(output$main$trv_type)
```

The above should report factor as the class

**Comment by allaway** [comment]
Hmm. I tried the above (using the test data) and `output` is `NULL`, so `class(output$main$trv_type)` is class `NULL` too. The plot still shows up in the plots pane (I'm using RStudio). 

I've attached the plot using the sample data and the snippet from above:
`cols <- c("blue", "red", "pink", "green", "orange", "black", "brown")
waterfall(brcaMAF, fileType = "MAF", mainRecurCutoff = 0.1, mainDropMut = TRUE, mainPalette = cols)`

![rplot](https://cloud.githubusercontent.com/assets/20408008/25759993/a0070942-3189-11e7-824e-6a081c2e75a2.png)

**Comment by allaway** [comment]
@zlskidmore I just wanted to follow up on this. I was plotting with `mainDropMut = TRUE`. What I realized is if I took into account all 18 mutation types in making my color vector, the colors would map correctly. 

eg. `cols <- c("#CC2936", "#397367", "#28536B", "black", "black", "black", "black", "#A7BBEC", "#0D1B2A", "black", "black", "black", "black", "black", "black", "black", "black", "black")`

where the black mutations will not be graphed (they are not in my data). In summary, when I list 18 colors, I can get everything to map discretely. 

I still have the issue I mentioned above, where the output is `NULL` when I try to store it, even though I see the plot.

Cheers, Robert
